presto-main/src/main/java/com/facebook/presto/sql/planner/SystemPartitioningHandle.java

```diff
@@ -110,6 +110,18 @@ public boolean isCoordinatorOnly()
         return partitioning == SystemPartitioning.COORDINATOR_ONLY;
     }
 
+    @Override
+    public boolean isBroadcast()
+    {
+        return function == SystemPartitionFunction.BROADCAST;
+    }
+
+    @Override
+    public boolean isArbitrary()
+    {
+        return function == SystemPartitionFunction.ROUND_ROBIN;
```

Comment:
Does this intentionally exclude ARBITRARY_DISTRIBUTION which has SystemPartitioningFunction.UNKNOWN?  makes the name a bit confusing.

Reply:
`ARBITRARY_DISTRIBUTION` looks like a hack used by the `AssignUniqueId`. Basically it is used to say `uniqueId`'s are grouped together, but the function used to group them together is "unknown".

This partitioning is never used to actually partition data, as the rule how to partition is not defined. It is used only to state as a matter of fact that something is partitioned based on a certain set of keys, but partitioning function is unknown.